### PR TITLE
Docker-image-pusher

### PR DESCRIPTION
**Pull request by FredrikFolkesson**
Pushes images from one repo to another.

### CODE REVIEW

File: latest-qualified-docker-tag.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const MS_PER_DAY = 24 * 60 * 60 * 1000
+
+async function main() {
+    try {
+        await dockerHubAPI.login(process.env.DH_USER, process.env.DH_PASS)
+        const [repo, image] = process.argv[2].split("/");
+        const tags = await dockerHubAPI.tags(repo, image);
+
+        const latestQualifiedImage = tags.filter(tag => tag.name.match(/^\d+\.\d+\.\d+$/) && semver.gt(tag.name, process.argv[3]) && Date.parse(tag.last_updated) > (Date.now() - MS_PER_DAY * process.env.NEWNESS_THRESHOLD))
```
Comment (carlioth) [inline latest-qualified-docker-tag.js]: A comment above what this magic match will do. My guess is that you only would like to select dockerhub images that has a proper versioning like: 12.249.0 and not 12.250.0-dev.52-ebf6347

File: new-image-pusher.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    if [ "$tag" == "undefined" ]; then
+      echo "No newer, valid image found for $from_image, ignoring."
+    else
+      echo "Found image $from_image:$tag will push to dockerhub as $to_image:$tag"
+      docker pull $from_image:$tag >/dev/null
+      docker tag $from_image:$tag $to_image:$tag >/dev/null
+      docker push $to_image:$tag
+
+      #cleanup
+      docker image rm $from_image:$tag >/dev/null
```
Comment (carlioth) [inline new-image-pusher.sh]: We don't have anything else on the machine running this script that will be dependent on one of the images that you will remove here?

Comment (carlioth) [reply new-image-pusher.sh]: perhaps go for the `docker image prune` that will remove unused images

File: latest-qualified-docker-tag.js (hunk 1 above)
Comment (FredrikFolkesson) [reply latest-qualified-docker-tag.js]: will add!

File: new-image-pusher.sh (hunk 2 above)
Comment (FredrikFolkesson) [reply new-image-pusher.sh]: We should not have anything actually running these images we are bumping.

Will prune remove an image that was just added?

Comment (carlioth) [reply new-image-pusher.sh]: I think prune will remove all images that don't have a running container associated. So the new will be removed yes

**Pull request merged**